mmai-cleanup.sh

```diff
@@ -0,0 +1,496 @@
+    kcd "-n ""$NAMESPACE"" ${KIND}.$(printapiversion "$APIVERSION") ""$NAME"""
+  done
+
+  kcdns "$NS"
+done
+
+for NS in $(kubectl get namespace --no-headers -o custom-columns=NAME:.metadata.name | grep "^c-"); do
+  kubectl get "$(kubectl api-resources --namespaced=true --verbs=delete -o name| grep -v events\.events\.k8s\.io | grep -v ^events$ | tr "\n" "," | sed -e 's/,$//')" -n "$NS" --no-headers -o custom-columns=NAME:.metadata.name,NAMESPACE:.metadata.namespace,KIND:.kind,APIVERSION:.apiVersion | while read -r NAME NAMESPACE KIND APIVERSION; do
+    kcpf -n "$NAMESPACE" "${KIND}.$(printapiversion "$APIVERSION")" "$NAME"
+    kcd "-n ""$NAMESPACE"" ${KIND}.$(printapiversion "$APIVERSION") ""$NAME"""
+  done
+
+  kcdns "$NS"
+done
+
+for NS in $(kubectl get namespace --no-headers -o custom-columns=NAME:.metadata.name | grep "^user-"); do
+  kubectl get "$(kubectl api-resources --namespaced=true --verbs=delete -o name| grep -v events\.events\.k8s\.io | grep -v ^events$ | tr "\n" "," | sed -e 's/,$//')" -n "$NS" --no-headers -o custom-columns=NAME:.metadata.name,NAMESPACE:.metadata.namespace,KIND:.kind,APIVERSION:.apiVersion | while read -r NAME NAMESPACE KIND APIVERSION; do
+    kcpf -n "$NAMESPACE" "${KIND}.$(printapiversion "$APIVERSION")" "$NAME"
+    kcd "-n ""$NAMESPACE"" ${KIND}.$(printapiversion "$APIVERSION") ""$NAME"""
+  done
+
+  kcdns "$NS"
+done
+
+for NS in $(kubectl get namespace --no-headers -o custom-columns=NAME:.metadata.name | grep "^u-"); do
+  kubectl get "$(kubectl api-resources --namespaced=true --verbs=delete -o name| grep -v events\.events\.k8s\.io | grep -v ^events$ | tr "\n" "," | sed -e 's/,$//')" -n "$NS" --no-headers -o custom-columns=NAME:.metadata.name,NAMESPACE:.metadata.namespace,KIND:.kind,APIVERSION:.apiVersion | while read -r NAME NAMESPACE KIND APIVERSION; do
+    kcpf -n "$NAMESPACE" "${KIND}.$(printapiversion "$APIVERSION")" "$NAME"
+    kcd "-n ""$NAMESPACE"" ${KIND}.$(printapiversion "$APIVERSION") ""$NAME"""
+  done
+
+  kcdns "$NS"
+done
+
+# Delete logging CRDs
+for CRD in $(kubectl get crd -o name | grep logging\.banzaicloud\.io); do
+  kcd "$CRD"
+done
+
+# Delete monitoring CRDs
+for CRD in $(kubectl get crd -o name | grep monitoring\.coreos\.com); do
+  kcd "$CRD"
+done
+
+# Delete OPA CRDs
+for CRD in $(kubectl get crd -o name | grep gatekeeper\.sh); do
+  kcd "$CRD"
+done
+
+# Delete Istio CRDs
+for CRD in $(kubectl get crd -o name | grep istio\.io); do
+  kcd "$CRD"
+done
+
+# Delete cluster-api CRDs
+for CRD in $(kubectl get crd -o name | grep cluster\.x-k8s\.io); do
+  kcd "$CRD"
+done
+
+# Delete all cattle CRDs
+# Exclude helm.cattle.io and addons.k3s.cattle.io to not break RKE2 addons
+for CRD in $(kubectl get crd -o name | grep cattle\.io | grep -v helm\.cattle\.io | grep -v k3s\.cattle\.io); do
+  kcd "$CRD"
+done
+
+# Delete all mmai CRDs
+for CRD in $(kubectl get crd -o name | grep -e mmai -e kueue -e nvidia -e mmcloud); do
+  kcd "$CRD"
```

Comment:
I found that this is not enough. The finalizers on the CRs of these CRD types should be removed, otherwise the CRD deletion would stuck.